pkg/sidecarlogresults/sidecarlogresults.go

```diff
@@ -0,0 +1,179 @@
+	Name  string
+	Value string
+}
+
+func (result *SidecarLogResult) log() {
+	res, _ := json.Marshal(&result)
+	fmt.Println(string(res))
+}
+
+// LookForResults waits for results to be written out by the steps
+// in their results path and prints them in a structured way to its
+// stdout so that the reconciler can parse those logs.
+func LookForResults(resultsDir string, resultNames string) error {
+	expectedResults := sets.NewString()
+	for _, s := range strings.Split(resultNames, ",") {
+		expectedResults.Insert(s)
+	}
+	resultsFetched := sets.NewString()
+	results := map[string]string{}
+	resultParsed := map[string]bool{}
+	startTimesPerResult := map[string]int64{}
+	millisecondsTowaitForEmptyResult := int64(1000)
+	sleepTime := time.Millisecond
+	for true {
+		f, err := os.Open(resultsDir)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		files, err := f.Readdir(0)
+		if err != nil {
+			return err
+		}
+		for _, v := range files {
+			if _, ok := resultParsed[v.Name()]; !ok {
+				resultParsed[v.Name()] = false
+			}
+			if expectedResults.Has(v.Name()) && !resultsFetched.Has(v.Name()) {
+				value, err := os.ReadFile(fmt.Sprintf("%v/%v", resultsDir, v.Name()))
+				if err != nil {
+					return err
+				}
+				newResult := SidecarLogResult{
+					Name:  v.Name(),
+					Value: string(value),
+				}
+				if val, ok := results[newResult.Name]; ok {
+					if val == newResult.Value {
+						if newResult.Value == "" {
+							now := time.Now().UnixMilli()
+							if _, ok := startTimesPerResult[newResult.Name]; !ok {
+								startTimesPerResult[newResult.Name] = now
+							} else if now-startTimesPerResult[newResult.Name] >= millisecondsTowaitForEmptyResult {
+								// value is still empty after second from the time that the result file was created.
+								resultParsed[newResult.Name] = true
+							}
+						} else {
+							// populated value is not changing anymore after 50 ms.
+							// assume that the contents have been fully written.
+							resultParsed[newResult.Name] = true
+						}
+					}
+				}
+				results[newResult.Name] = newResult.Value
+
+				if resultParsed[newResult.Name] == true {
+					newResult.log()
+					resultsFetched.Insert(v.Name())
+				}
+			}
+		}
+		f.Close()
+		time.Sleep(sleepTime)
+		if reflect.DeepEqual(resultsFetched, expectedResults) {
+			break
+		}
+	}
+	return nil
+}
+
+// GetResultsFromSidecarLogs extracts results from the logs of the results sidecar
+func GetResultsFromSidecarLogs(ctx context.Context, clientset kubernetes.Interface, namespace string, name string, container string, sidecarLogResults []v1beta1.PipelineResourceResult) ([]v1beta1.PipelineResourceResult, error) {
+	podLogOpts := corev1.PodLogOptions{Container: container}
+	req := clientset.CoreV1().Pods(namespace).GetLogs(name, &podLogOpts)
+	sidecarLogs, err := req.Stream(ctx)
+	if err != nil {
+		podInitializingError := fmt.Sprintf(`container "%v" in pod "%v" is waiting to start: PodInitializing`, container, name)
```

Comment:
This is going to be brittle. Stream documentation says this will return an [APIStatus](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#APIStatus) if possible - let's check against the status state instead - i.e. status.Reason == [PodInitializing](https://pkg.go.dev/k8s.io/kubernetes/pkg/kubelet#PodInitializing)

Reply:
I'm getting a nil type not API status: 
```
if err != nil {
                podInitializingError := fmt.Sprintf(`container "%v" in pod "%v" is waiting to start: PodInitializing`, container, name)
                if err.Error() == podInitializingError {
                        log.Println("========)
                        log.Println(sidecarLogs)
                        log.Println(err)
                        log.Println("========")
                        return sidecarLogResults, nil
                }
                return sidecarLogResults, err
        }
```

```
2022/11/22 15:54:47 ========                                                                                                       
2022/11/22 15:54:47 <nil>                                                                                                           
2022/11/22 15:54:47 container "sidecar-results" in pod "larger-results-gw6xv-pod" is waiting to start: PodInitializing              
2022/11/22 15:54:47 ========   
```

How do I access the API Status?

Reply:
Nevermind. I figured it out. I need to get the pod and then check its status. I fixed it 😄